dms/models/ir_attachment.py

```diff
@@ -0,0 +1,59 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _get_dms_directories(self, res_model, res_id):
+        return self.env["dms.directory"].search(
+            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        )
+
+    @api.model
+    def create(self, vals):
+        if "dms_file" in vals and vals["dms_file"]:
+            del vals["dms_file"]
+            return super(IrAttachment, self).create(vals)
+        else:
+            attachment_id = super(IrAttachment, self).create(vals)
+            if attachment_id.res_model and attachment_id.res_id:
+                directory_ids = self._get_dms_directories(
+                    attachment_id.res_model, attachment_id.res_id
+                )
+                if not directory_ids:
```

Comment:
Please try to improve this ugly code with so many cyclomatic complexity and bad variable names.